torch/csrc/jit/codegen/cuda/manager.cpp

```diff
@@ -119,10 +154,48 @@ void runCudaFusionGroup(const Node* const fusion_node, Stack& stack) {
   int32_t kernel_id = fusion_node->i(attr::cache_id);
 
   // Currently we just construct I/O tensors for static graph;
-  const std::shared_ptr<Graph> graph = fusion_node->g(attr::Subgraph);
+  std::shared_ptr<Graph> graph = fusion_node->g(attr::Subgraph);
+
   const auto nInputs = graph->inputs().size();
   at::ArrayRef<IValue> inputs = last(stack, nInputs);
 
+  // shape inference in graph
```

Comment:
Uncertain what you're doing here to the end of this file. Would be helpful to chat about it.

Reply:
Brief idea, we pop the items from the stack and take them as inputs to our node. We push the output back so future operation can use it. This is expected from interpreter.

We can chat off line as well.